subprojects/docs/src/docs/release/notes.md

```diff
@@ -148,6 +148,17 @@ This avoids a 2nd request for the JAR file when the POM is missing, making depen
 
 You can also use this to programmatically opt-in to using the new Gradle metadata format for one repository using `metadataSources { gradleMetadata() }`.
 
+### Specifying reasons for dependency declaration and resolution rules
```

Comment:
I'd try to indicate that the "reason" is provided by the user e.g. by saying "Ability to specify custom reasons for dependency declaration and resolution rules".

Reply:
Good suggested. I changed it.